src/gc/env/gcenv.base.h

```diff
@@ -231,23 +231,21 @@ inline uint8_t BitScanForward(uint32_t *bitIndex, uint32_t mask)
 #ifdef _MSC_VER
     return _BitScanForward((unsigned long*)bitIndex, mask);
 #else // _MSC_VER
-    unsigned char ret = FALSE;
     int iIndex = __builtin_ffsl(mask);
-    if (iIndex != 0)
-    {
-        *bitIndex = (uint32_t)(iIndex - 1);
-        ret = TRUE;
-    }
-
-    return ret;
+    *bitIndex = static_cast<uint32_t>(iIndex - 1);
```

Comment:
Note: There is a difference between what this implementation and the Windows implementation returns in the bitIndex for the case of mask being zero. The Windows implementation sets the bitIndex to 0, this one sets it to 0xffffffff. I don't think it is a problem at all, but it would be great to mention that in the comment on the function.